scripts/nav-tree-utils.js

```diff
@@ -0,0 +1,84 @@
+import { createOptimizedPicture } from './lib-franklin.js';
+
+function addBackdropEventListeners(element) {
+  const backdrop = document.querySelector('.backdrop');
+  element.addEventListener('mouseover', () => {
+    backdrop.classList.add('visible');
+  });
+  element.addEventListener('mouseleave', () => {
+    backdrop.classList.remove('visible');
+  });
+}
+
+/**
+ * @param {String} HTML representing a single element
+ * @return {Element}
+ */
+function htmlToElement(html) {
+  const template = document.createElement('template');
+  const trimmedHtml = html.trim(); // Never return a text node of whitespace as the result
+  template.innerHTML = trimmedHtml;
+  return template.content.firstChild;
+}
+
+function decorateChildNodes(parent, json, level) {
+  const nodes = json.reduce((accumalator, data) => {
+    if (data.parent?.toLowerCase() === parent.category.toLowerCase()) {
+      const children = data.hasChild === 'true' ? decorateChildNodes(data, json, level + 1) : '';
+      if (data.link && !children) {
+        return `${accumalator} <a class="link" href=${data.link}>${data.category}</a>`;
+      } if (!data.link && children) {
+        return `${accumalator} <div class="menu-level-${level}-item"><h6 class="subtitle">${data.category}</h6>${children}</div>`;
+      }
+      return accumalator;
+    }
+    return accumalator;
+  }, '');
+  return `<div class="menu-level-${level}">${nodes}</div>`;
+}
+
+function decorateNodes(json, level) {
+  const ul = htmlToElement(`<ul class=menu-level-${level}></ul>`);
+  json.forEach((data) => {
+    if (!data.parent || data.parent === '') {
+      const children = data.hasChild === 'true' ? decorateChildNodes(data, json, level + 1) : '';
+      let li;
+      if (children) {
+        const picture = createOptimizedPicture(data.image, '', false, [{ width: '800' }]);
+        li = htmlToElement(`<li class="drop menu-level-${level}-item"> 
+          <a class="link" href=${data.link}>${data.category}</a>
```

Comment:
The `data.link` here and everywhere should be governed by the page domain. For e.g. `hlx.live` should have links with `hlx.live` domain and similarly for preview and prod domains.